code/modules/hydroponics/biogenerator.dm

```diff
@@ -148,11 +184,20 @@
 	else
 		to_chat(user, span_warning("You cannot put this in [src.name]!"))
 
-/obj/machinery/biogenerator/AltClick(mob/living/user)
+/obj/machinery/biogenerator/attack_hand_secondary(mob/user, list/modifiers)
 	. = ..()
-	if(user.canUseTopic(src, BE_CLOSE, FALSE, NO_TK) && can_interact(user))
-		eject_beaker(user)
+	if(. == SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN)
+		return
+	if(!can_interact(user) || !user.canUseTopic(src, !issilicon(user), FALSE, NO_TK))
+		return
+	eject_beaker(user)
+	return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+
+/obj/machinery/biogenerator/attack_robot_secondary(mob/user, list/modifiers)
+	return attack_hand_secondary(user, modifiers)
 
+/obj/machinery/biogenerator/attack_ai_secondary(mob/user, list/modifiers)
+	return attack_hand_secondary(user, modifiers)
```

Comment:
This makes AI/Cyborgs able to eject beakers from anywhere and have it appear wherever they are, no?

Reply:
From what I have tried with the other chemistry machines and my PR, it drops the beaker on top of the machine.
But I can do further tests later, specially with TK now that you mention it.

Again I wanted to make the biogenerator's controls similar with the rest of chem machines.